Feat: Add Support for OCI

theBeginner86 opened the pull request:
**Description**

Adds functionality to:
1. Compress into OCI Artifacts
2. De-Compress OCI Artifacts


This would be helpful for importing/exporting Designs and Models.

This PR fixes #

**Notes for Reviewers**


**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
4. Build and test your changes before submitting a PR. 
5. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

leecalcote commented on models/oci/oci.go:
```diff
@@ -0,0 +1,145 @@
+
+	img, err = mutate.Append(img, mutate.Addendum{Layer: layer})
+	if err != nil {
+		return nil, fmt.Errorf("appeding content to artifact failed: %w", err)
```
```suggestion
		return nil, fmt.Errorf("appending content to artifact failed: %w", err)
```

leecalcote commented on models/oci/build.go:
```diff
@@ -0,0 +1,146 @@
+)
+
+
+func build(artifactPath, sourceDir string, ignorePaths []string) (err error) {
```
```suggestion
// Supports local folder (directory) path only. 
func build(artifactPath, sourceDir string, ignorePaths []string) (err error) {
```

leecalcote commented:
@MUzairS15 you're going to want to review this.

MUzairS15 reviewed (commented):
`errors`  needs to be transformed into `Meshkit errors`.
I wonder if it's better to directly import theire packages or else if we are directly sourcing form their repo, then appropriate licensing needs to be added in the code.
// @theBeginner86

theBeginner86 commented:
> `errors` needs to be transformed into `Meshkit errors`. I wonder if it's better to directly import theire packages or else if we are directly sourcing form their repo, then appropriate licensing needs to be added in the code. // @theBeginner86

Added MeshKit errors.
Removed duplicated code and imported the necessary functions.

leecalcote commented:
Re-running build...

leecalcote commented:
This is ready, right, @theBeginner86?

leecalcote commented:
I tried, but...

theBeginner86 commented:
> Build failure...

Fixed: https://github.com/meshery/meshkit/actions/runs/7462230856/job/20304075337?pr=443
